src/main/java/io/jenkins/plugins/folderauth/FolderAuthorizationStrategyManagementLink.java

```diff
@@ -207,6 +269,67 @@ public void doAssignSidToFolderRole(@QueryParameter(required = true) String role
         redirect();
     }
 
+    /**
+     * Removes {@code sid} from the folder role identified by {@code roleName}.
+     *
+     * @param roleName the name of the global to which {@code sid} will be assigned to.
+     * @param sid      the sid of the user/group to be assigned.
+     * @throws IllegalStateException    when {@link Jenkins#getAuthorizationStrategy()} is
+     *                                  not {@link FolderBasedAuthorizationStrategy}
+     * @throws IllegalArgumentException when no role with name equal to {@code roleName} exists.
+     */
+    @RequirePOST
+    @Restricted(NoExternalUse.class)
+    public void doRemoveSidFromFolderRole(@QueryParameter(required = true) String roleName,
+                                          @QueryParameter(required = true) String sid) {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        FolderAuthorizationStrategyAPI.removeSidFromFolderRole(sid, roleName);
+        redirect();
+    }
+
+    /**
+     * Adds an {@link AgentRole} to {@link FolderBasedAuthorizationStrategy}.
+     *
+     * @param request the request to create the role
+     * @throws IllegalStateException when {@link Jenkins#getAuthorizationStrategy()} is
+     *                               not {@link FolderBasedAuthorizationStrategy}
+     */
+    @RequirePOST
+    @Restricted(NoExternalUse.class)
+    public void doAddAgentRole(@JsonBody AgentRoleCreationRequest request) {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        FolderAuthorizationStrategyAPI.addAgentRole(request.getAgentRole());
+    }
+
+    /**
+     * Assigns or removes a {@code sid} to the agent role identified by {@code roleName}.
+     * Does nothing if the {@code roleName} does not exist.
+     *
+     * @param roleName the name of the agent role to which {@code sid} will be assigned to.
+     * @param sid      the sid of the user/group to be assigned.
+     *                 @param action   the action to be performed ({@code assign} or {@code remove})
+     * @throws IllegalStateException            when {@link Jenkins#getAuthorizationStrategy()} is
+     *                                          not {@link FolderBasedAuthorizationStrategy}
+     * @throws IllegalArgumentException when {@code action} is neither {@code assign} nor {@code remove}
+     */
+    @RequirePOST
+    @Restricted(NoExternalUse.class)
+    public void doAssignOrRemoveSidAgentRole(@QueryParameter(required = true) String roleName,
```

Comment:
This looks like a lot of boilerplate for a simple task. Perhaps we can move it to JavaScript. See the other review comment.

Reply:
ok i can do a quick refactor to combine the two methods if its preferred